### PR TITLE
IODEMO: add listen retry for server

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
add listen retry for iodemo-server

## Why ?
cherry-picked from https://github.com/yosefe/ucx/pull/94

### CODE REVIEW

**Review by yosefe — APPROVED** [review]

File: test/apps/iodemo/ucx_wrapper.h [hunk 1]
```diff
@@ -86,6 +86,9 @@ class UcxContext {
 
     void progress();
 
+    static const std::string sockaddr_str(const struct sockaddr* saddr,
```
Comment (brminich) [inline test/apps/iodemo/ucx_wrapper.h:89]: why is this move needed?

Comment (evgeny-leksikov) [reply test/apps/iodemo/ucx_wrapper.h:89]: to make it public and call outside of wrapper from io_demo

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

**Pull request merged**